src/pages/projects/projects-list/ProjectsList.tsx

```diff
@@ -1,39 +1,35 @@
 import { useEffect, useState } from 'react';
 import { Project } from '@model';
 import { ProjectTile } from './project-tile/ProjectTile';
-import { TextInput } from '@components';
 import styles from './ProjectsList.module.less';
 
 interface Props {
-  projects: Project[]
+  projects: Project[],
+  filter: string
 }
 
 const ProjectsList = (props: Props) => {
-  const [searchQuery, setSearchQuery] = useState<string>('');
   const [queryResult, setQueryResult] = useState<Project[]>([]);
 
   useEffect(() => {
-    setQueryResult(props.projects.filter((project: Project) => project.name.indexOf(searchQuery) !== -1));
-  }, [searchQuery, props.projects]);
+    setQueryResult(props.projects.filter((project: Project) => {
+      const pattern = props.filter.toLocaleLowerCase();
+      const text = project.name.toLocaleLowerCase();
+      let it = 0;
+      for(let i=0; i<text.length; i++)
+        if(it < pattern.length && text[i] === pattern[it])
+          it++;
+      return it === pattern.length;
+    }));
+  }, [props.filter, props.projects]);
```

Comment:
would you try whether `useMemo` works here? It should

```
const queryResult = useMemo(() => {
   return props.projects.filter((project: Project) => {
      const pattern = props.filter.toLocaleLowerCase();
      const text = project.name.toLocaleLowerCase();
      let it = 0;
      for(let i=0; i<text.length; i++)
        if(it < pattern.length && text[i] === pattern[it])
          it++;
      return it === pattern.length;
    }));
}, [props.filter, props.project]);
```

Reply:
Good point, applied